### PR TITLE
Onboarding - Add a new flow for readymade templates

### PR DESCRIPTION
**Pull request by BogdanUngureanu**
Adds a new onboarding flow for the readymade templates, including the Launchpad integration.

Related to #https://github.com/Automattic/dotcom-forge/issues/7814 and https://github.com/Automattic/dotcom-forge/issues/7815

## Proposed Changes
Adds a new onboarding flow for Readymade templates that:
* Sets the Assembler theme on the site
* Applies the Readymade template
* Redirects the user to the Site Editor
* When the user exists the SE, the launchpad is triggered for this specific flow

## Why are these changes being made?

* Needed in the context of Readymade templates in Patterns Library.

## Testing Instructions

* Apply the WPCOM changes from [this](https://github.com/Automattic/jetpack/pull/37989) PR to your sandbox: `bin/jetpack-downloader test jetpack-mu-wpcom-plugin add/readymade-template-launchpad-task-list`
* Use the live link and go to `/setup/readymade-template?readymadeTemplateId=1`
* This will automatically create a site for you
* Will now land in the site-editor
* Exit the Site Editor and notice that you land in the launchpad
* Notice that the launchpad URL is `setup/readymade-template/launchpad`
* Click on the domains tab
* You should be able to change your WordPress.com subdomain besides registering a new domain
* Try out other tabs and see if they are working fine (except for the Write post that sends you to production launchpad - causing a redirect since the flow doesn't exist there)

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by BogdanUngureanu** [comment]
Pushed a commit that fixes the problem. The PR is RfR again.

**Review by xavier-lc — APPROVED** [review]
Works as described 👌

**Comment by mmtr** [comment]
Looks promising!

Do you know why the header ends up with this "About / Learn more" menu instead of the social menu that is supposed to have the template?

This PR | Expected menu
--- | ---
Screenshot 2024-06-25 at 14 55 54 | Screenshot 2024-06-25 at 14 58 49

File: client/landing/stepper/declarative-flow/readymade-template.tsx [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+					/**
+					 * @todo Separate the content in 3 sections. Ideally the entire template configuration should be done one the server, see the above comment.
+					 *
+					 * For now we piggyback on Site Assembler's API endpoint to apply the template on the site.
+					 */
+					homeHtml: readymadeTemplate.content,
+					headerHtml: '',
+					footerHtml: '',
+					pages: [],
+					globalStyles: {},
```
Comment (dsas) [inline client/landing/stepper/declarative-flow/readymade-template.tsx:316]: Something should apply the style variation that the selected design had active. It looks like this is the right place?

File: client/landing/stepper/declarative-flow/readymade-template.tsx [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+					canReplaceContent: true,
+					// All sites using the assembler set the option wpcom_site_setup
+					siteSetupOption: 'assembler',
```
Comment (dsas) [inline client/landing/stepper/declarative-flow/readymade-template.tsx]: This doesn't use the assembler though 🤔 I'm not sure setting this to assembler is right

Comment (dsas) [reply client/landing/stepper/declarative-flow/readymade-template.tsx]: Looking at PCYsg-K6z-p2 I think we need to use a new string here

File: client/landing/stepper/declarative-flow/readymade-template.tsx (hunk 1 above)
Comment (mmtr) [reply client/landing/stepper/declarative-flow/readymade-template.tsx:316]: I created a separate issue for this: https://github.com/Automattic/dotcom-forge/issues/7944

File: client/landing/stepper/declarative-flow/readymade-template.tsx (hunk 2 above)
Comment (Copons) [reply client/landing/stepper/declarative-flow/readymade-template.tsx]: +1

Let's use `readymade-template` to distinguish. We can revisit before the (real) launch, if needed.

The `assembler` option should only be used by the Assembler **flow**, not the theme. 🙂 
Furthermore, it's only for filtering these sites; AFAIK, this option has no functional meaning.

**Pull request merged**